src/main/java/muramasa/antimatter/datagen/providers/AntimatterBlockStateProvider.java

```diff
@@ -87,13 +101,37 @@ public void state(Block block, ResourceLocation... textures) {
             simpleBlock(block, getSimpleModel(block, textures[0]));
         } else if (textures.length == 2) {
             simpleBlock(block, getLayeredModel(block, textures[0], textures[1]));
+        } else if (textures.length == 6){
+            horizontalBlock(block, getSixSidedSimpleModel(block, false, textures));
+        } else if (textures.length == 12){
+            horizontalBlock(block, getSixSidedLayeredModel(block, false, textures));
         }
     }
 
+    @Override
+    public void horizontalBlock(Block block, Function<BlockState, ModelFile> modelFunc) {
+        getVariantBuilder(block)
+                .forAllStates(state -> ConfiguredModel.builder()
+                        .modelFile(modelFunc.apply(state))
+                        .rotationY((state.has(BlockStateProperties.HORIZONTAL_FACING) ? (int) state.get(BlockStateProperties.HORIZONTAL_FACING).getHorizontalAngle() : (int) Direction.SOUTH.getHorizontalAngle()) % 360)
+                        .build()
+                );
+    }
+
     public BlockModelBuilder getSimpleModel(Block block, ResourceLocation texture) {
         return getBuilder(block).parent(models().getExistingFile(loc(Ref.ID, "block/preset/simple"))).texture("all", texture);
     }
 
+    public BlockModelBuilder getSixSidedSimpleModel(Block block, boolean active, ResourceLocation... texture) {
+        BlockModelBuilder builder = active ? getActiveBuilder(block) : getBuilder(block);
+        return builder.parent(models().getExistingFile(loc(Ref.ID, "block/preset/simple"))).texture("down", texture[0]).texture("up", texture[1]).texture("south", texture[2]).texture("north", texture[3]).texture("west", texture[4]).texture("east", texture[5]).texture("particle", texture[1]);
+    }
+
+    public BlockModelBuilder getSixSidedLayeredModel(Block block, boolean active, ResourceLocation... texture) {
+        BlockModelBuilder builder = active ? getActiveBuilder(block) : getBuilder(block);
+        return builder.parent(models().getExistingFile(loc(Ref.ID, "block/preset/layered"))).texture("basedown", texture[0]).texture("baseup", texture[1]).texture("basesouth", texture[2]).texture("basenorth", texture[3]).texture("basewest", texture[4]).texture("baseeast", texture[5]).texture("overlaydown", texture[6]).texture("overlayup", texture[7]).texture("overlaysouth", texture[8]).texture("overlaynorth", texture[9]).texture("overlaywest", texture[10]).texture("overlayeast", texture[11]).texture("particle", texture[1]);
+    }
+
```

Comment:
Rename these to `getSimpleModel` and `getLayeredModel` respectively, and remove the active param

Reply:
Scrap `getSixSidedSimpleModel` altogether (use `ModelProvider#cube`), I believe that's pretty much the same? You may have to add a `particle` property but I think vanilla does that by default?